packages/codegen-ui-react/lib/forms/form-renderer-helper/form-state.ts

```diff
@@ -240,47 +241,49 @@ export const resetStateFunction = (fieldConfigs: Record<string, FieldConfigMetad
   const recordOrInitialValues = recordName ? 'cleanValues' : 'initialValues';
 
   const stateNames = new Set<string>();
-  const expressions = Object.entries(fieldConfigs).reduce<Statement[]>(
-    (acc, [name, { isArray, sanitizedFieldName, componentType, dataType }]) => {
-      const stateName = name.split('.')[0];
-      const renderedName = sanitizedFieldName || stateName;
-      if (!stateNames.has(stateName)) {
-        const accessExpression = isValidVariableName(stateName)
-          ? factory.createPropertyAccessExpression(
-              factory.createIdentifier(recordOrInitialValues),
-              factory.createIdentifier(stateName),
-            )
-          : factory.createElementAccessExpression(
-              factory.createIdentifier(recordOrInitialValues),
-              factory.createStringLiteral(stateName),
-            );
+  const expressions = Object.entries(fieldConfigs).reduce<Statement[]>((acc, [name, fieldConfig]) => {
+    const { isArray, sanitizedFieldName, componentType, dataType } = fieldConfig;
+    const stateName = name.split('.')[0];
+    const renderedName = sanitizedFieldName || stateName;
+    if (!stateNames.has(stateName)) {
+      const accessExpression = isValidVariableName(stateName)
+        ? factory.createPropertyAccessExpression(
+            factory.createIdentifier(recordOrInitialValues),
+            factory.createIdentifier(stateName),
+          )
+        : factory.createElementAccessExpression(
+            factory.createIdentifier(recordOrInitialValues),
+            factory.createStringLiteral(stateName),
+          );
```

Comment:
not for this pr but ive been noticing we've been replicating this logic in a lot of places 
i wonder we should unify it into one function which simply returns the variable name

Reply:
I just created a separate helper in this PR. I missed using it here. Will rev in the next PR